include/deal.II/base/quadrature_point_data.h

```diff
@@ -0,0 +1,808 @@
+   * This vector will contain all scalar and/or Tensorial data local to each
+   * quadrature point.
+   *
+   * @p values will be resized to number_of_values() .
+   */
+  virtual void pack_values(std::vector<double> &values) const = 0;
+
+  /**
+   * The opposite of the above, namely
+   * unpack a vector @p values into the data stored in this class.
+   *
+   * @p values will be resized to number_of_values() .
+   */
+  virtual void unpack_values(const std::vector<double> &values) = 0;
+};
+
+
+#ifdef DEAL_II_WITH_P4EST
+namespace distributed
+{
+  /**
+   * A class for the transfer of continuous data stored at quadrature points when
+   * performing h-adaptive refinement of parallel::distributed::Triangulation .
+   *
+   * <h3>Implementation details</h3>
+   *
+   * This class implements the transfer of the quadrature point data between
+   * cells in case of adaptive refinement. That also includes automatic shipping of
+   * information between different processors.
+   *
+   * To that end, the constructor of the class is provided with three main objects:
+   * scalar FiniteElement @projection_fe, @p mass_quadrature and @p data_quadrature
+   * Quadrature rules.
+   * First, the data located at @p data_quadrature of each cell is L2-projected
+   * to the continuous space defined by a single FiniteElement @projection_fe .
+   * In  doing so the mass matrix of this element is required, which will be calculated
+   * with the @p mass_quadrature rule . Should the cell now belong to another processor,
+   * the data is then sent to this processor. The class makes use of a feature of p4est
+   * (and parallel::distributed::Triangulation) that allows one to attach information
+   * to cells during mesh refinement and rebalancing.
+   * On receiving information on the target cell, the data is projected back to the
+   * quadrature points. In the case that local refinement is performed, we have to first
+   * project local DoF values of the parent element to each child.
+   *
+   * This class is templated by @p DataType type, however the user's @p DataType class
+   * has to be derived from the TransferableQuadraturePointData class. In practice
+   * that amounts to implementing the following three functions shown below for a
+   * quadrature point data with 2 scalars :
+   * <code>
+   * class MyQData : public TransferableQuadraturePointData
+   * {
+   * public:
+   *   unsigned int number_of_values() const
+   *   {
+   *      return 2;
+   *   }
+   *
+   *   // a function to pack scalars into a vector
+   *   void pack_values(std::vector<double> &values) const
+   *   {
+   *     Assert (scalars.size()==2, ExcInternalError());
+   *     scalars[0] = value1;
+   *     scalars[1] = value2;
+   *   }
+   *
+   *   void unpack_values(const std::vector<double> &values)
+   *   {
+   *     Assert (scalars.size() ==2, ExcInternalError());
+   *     value1 = scalars[0];
+   *     value2 = scalars[1];
+   *   }
+   * };
+   * </code>
```

Comment:
`@endcode`

I think in the example, you should actually declare the `value1,value2` members. Maybe give them more suggestive names so people can see how it would be used -- e.g., `elasticity_parameter_lambda` and `elasticity_parameter_mu`?